python/ray/tune/integration/pytorch_lightning.py

```diff
@@ -146,102 +154,58 @@ def _get_report_dict(self, trainer: Trainer, pl_module: LightningModule):
 
         return report_dict
 
-    def _handle(self, trainer: Trainer, pl_module: LightningModule):
-        report_dict = self._get_report_dict(trainer, pl_module)
-        if report_dict is not None:
-            tune.report(**report_dict)
-
-
-class _TuneCheckpointCallback(TuneCallback):
-    """PyTorch Lightning checkpoint callback
-
-    Saves checkpoints after each validation step.
-
-    .. note::
-        In Ray 2.4, we introduced
-        :class:`LightningTrainer <ray.train.lightning.LightningTrainer>`,
-        which provides native integration with PyTorch Lightning. Here is
-        :ref:`a simple example <lightning_mnist_example>` of how to use
-        ``LightningTrainer``.
-
-    Checkpoint are currently not registered if no ``tune.report()`` call
-    is made afterwards. Consider using ``TuneReportCheckpointCallback``
-    instead.
-
-    Args:
-        filename: Filename of the checkpoint within the checkpoint
-            directory. Defaults to "checkpoint".
-        on: When to trigger checkpoint creations. Must be one of
-            the PyTorch Lightning event hooks (less the ``on_``), e.g.
-            "train_batch_start", or "train_end". Defaults to "validation_end".
+    @contextmanager
+    def _get_checkpoint(
+        self, trainer: Trainer
+    ) -> Optional[Union[Checkpoint, LegacyCheckpoint]]:
+        if not self._save_checkpoints:
+            yield None
+            return
 
+        with tempfile.TemporaryDirectory() as checkpoint_dir:
+            trainer.save_checkpoint(os.path.join(checkpoint_dir, self._filename))
 
-    """
+            if _use_storage_context():
+                checkpoint = Checkpoint.from_directory(checkpoint_dir)
+            else:
+                checkpoint = LegacyCheckpoint.from_directory(checkpoint_dir)
```

Comment:
we can just use `train.Checkpoint` now -- the alias is now gated by the feature flag.